### PR TITLE
Improve symlink resolution

### PR DESCRIPTION
**Pull request by djvirgen**
Delegates symlink resolution to `Symlink#resolve` to allow chained resolvers to be executed.

This required writing a custom `merge()` function because lodash's `_.merge()` kills all prototype values when merging. I also looked at the popular `deep-extend` module, but it had weak support for arrays.

### CODE REVIEW

File: lib/Router.js [hunk 1]
```diff
@@ -425,26 +427,11 @@ Router.prototype.handle = function(request, connection) {
 
     return Promise.all(callbacks)
     .then(function(results) {
-        results = results.map(function(result) {
```
Comment (djvirgen) [inline lib/Router.js:428]: So glad to kill all this unnecessary code that was dealing with lodash's funky merge.

File: lib/Router.js [hunk 2]
```diff
@@ -425,26 +427,11 @@ Router.prototype.handle = function(request, connection) {
...
-                    // Convert to an object if an instance
-                    return (_.isObject(r)) ? _.merge({}, r) : r;
-                });
-            } else if (_.isObject(result)) {
-                return _.merge({}, result);
-            }
-
-            return result;
-        });
+        debug('got results from callbacks', results);
```
Comment (mariane-bekker) [inline lib/Router.js:430]: debug statement

File: lib/Router.js [hunk 3]
```diff
@@ -425,26 +427,11 @@ Router.prototype.handle = function(request, connection) {
...
 
         // Merges results from different functions
-        var result = _.merge.apply(null, results.concat([function Customizer(a, b) {
-            if (a instanceof Resource || b instanceof Resource) {
-                return _.merge(a, b);
-            }
-        }]));
+        var result = merge.apply(null, results);
+        debug('merged result from all callbacks', typeof result, result);
```
Comment (mariane-bekker) [inline lib/Router.js:434]: debug statement
 Add a line note

File: lib/Router.js (hunk 2 above)
Comment (djvirgen) [inline lib/Router.js:430]: These debug statements are intended to be left in. This is using the `debug` module, which defaults to a noop unless you start the node process with `DEBUG=monocle-api:*`

See https://github.com/visionmedia/debug

**Comment by mariane-bekker** [comment]
:+1:

**Pull request merged**